cypress/e2e/tests/migration/applicationinventory/analysis/auto_tagging_filter.test.ts

```diff
@@ -0,0 +1,125 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+/// <reference types="cypress" />
+
+import { login, getRandomApplicationData, getRandomAnalysisData } from "../../../../../utils/utils";
+import { Analysis } from "../../../../models/migration/applicationinventory/analysis";
+import { CredentialType, UserCredentials } from "../../../../types/constants";
+import * as data from "../../../../../utils/data_utils";
+import { CredentialsSourceControlUsername } from "../../../../models/administration/credentials/credentialsSourceControlUsername";
+import { Tag } from "../../../../models/migration/controls/tags";
+import { TagCategory } from "../../../../models/migration/controls/tagcategory";
+import { applicationTag } from "../../../../views/applicationinventory.view";
+import { SEC } from "../../../../types/constants";
+let source_credential;
+let tagCategory;
+let tag;
+
+describe(["@tier1"], "Source Analysis", () => {
+    before("Login", function () {
+        login();
+        source_credential = new CredentialsSourceControlUsername(
+            data.getRandomCredentialsData(
+                CredentialType.sourceControl,
+                UserCredentials.usernamePassword,
+                true
+            )
+        );
+        source_credential.create();
+
+        tagCategory = new TagCategory(
+            data.getRandomWord(8),
+            data.getColor(),
+            data.getRandomNumber(1, 30)
+        );
+        tagCategory.create();
+
+        tag = new Tag(data.getRandomWord(6), tagCategory.name);
+        tag.create();
+    });
+
+    beforeEach("Load data", function () {
+        cy.fixture("application").then(function (appData) {
+            this.appData = appData;
+        });
+        cy.fixture("analysis").then(function (analysisData) {
+            this.analysisData = analysisData;
+            this.techTags = analysisData["analysis_for_enableTagging"]["techTags"];
+        });
+        cy.intercept("GET", "/hub/application*").as("getApplication");
+    });
+
+    it("Apply search filter:Analysis on app details page", function () {
+        // Automates Polarion MTA-311
+        const application = new Analysis(
+            getRandomApplicationData("tackleTestApp_Source_autoTagging", {
+                sourceData: this.appData["tackle-testapp-git"]},
+                [tag.name],
+            ),
+            getRandomAnalysisData(this.analysisData["analysis_for_enableTagging"])
+        );
+        application.create();
+        cy.wait("@getApplication");
+        application.manageCredentials(source_credential.name, null);
+        application.analyze();
+        application.verifyAnalysisStatus("Completed");
+        cy.wait(2000);
+
+        application.filterTags("Analysis");
+        application.tagAndCategoryExists(this.techTags);
+        cy.get(applicationTag).should("not.contain", tag.name);
```

Comment:
@sshveta , I removed a few lines from application.tagAndCategoryExists() because I have to verify other things on the same page before closing it .